Tests/SpeziFHIRTests/FHIRStore/FHIRStoreObservationChangesTests.swift

```diff
@@ -0,0 +1,260 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2025 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import ModelsR4
+@testable import SpeziFHIR
+import XCTest
+
+
+final class FHIRStoreObservationChangesTests: XCTestCase {
+    private var store: FHIRStore! // swiftlint:disable:this implicitly_unwrapped_optional
+
+    
+    override func setUp() {
+        super.setUp()
+        store = FHIRStore()
+    }
+    
+    override func tearDown() {
+        store = nil
+        super.tearDown()
+    }
+    
+    @MainActor
+    func testChangesOnInsert() async throws {
+        let observation = try ModelsR4Mocks.createObservation()
+        let resource = FHIRResource(resource: observation, displayName: "Test Observation")
+
+        let observationsExpectation = expectation(description: "Observations should change")
+        let proceduresExpectation = expectation(description: "Procedures should NOT change")
+        proceduresExpectation.isInverted = true
+
+        withObservationTracking {
+            _ = store.procedures.count
+        } onChange: {
+            proceduresExpectation.fulfill()
+        }
+
+        withObservationTracking {
+            _ = store.observations.count
+        } onChange: {
+            observationsExpectation.fulfill()
+        }
```

Comment:
Great stuff, thank you very much for these tests and finding a small (performance) bug in the implementation! 🚀